src/with-props.js

```diff
@@ -0,0 +1,182 @@
+import { withRaw } from './with-raw';
+import {
+  debounce,
+  keys,
+  sym
+} from './util';
+import {
+  _updateDebounced,
+  defineProps,
+  normPropDefs,
+  syncAttributeToProperty
+} from './util/with-props';
+
+const _connected = sym();
+const _constructed = sym();
+
+const _observedAttributes = sym();
+const _prevProps = sym();
+const _props = sym();
+const _updateCallback = sym();
+const _updating = sym();
+
+export function withProps (Base = withRaw()) {
+  return class extends Base {
+    static get observedAttributes () {
+      const props = normPropDefs(this);
+      return keys(props)
+        .map(k => props[k].attribute)
+        .filter(Boolean)
+        .map(a => a.source)
+        .concat(this[_observedAttributes] || []);
+    }
+
+    static set observedAttributes (value) {
+      this[_observedAttributes] = value;
+    }
+
+    static get props () {
+      return { ...super.props, ...this[_props] };
+    }
+
+    static set props (value) {
+      this[_props] = value;
+    }
+
+    constructor () {
+      super();
+      if (this[_constructed]) return;
```

Comment:
There's a comment where it's defined with the other flag at the top. Polyfill pains again.